storage/tree_validation_test.go

```diff
@@ -162,8 +163,11 @@ func TestValidateTreeForCreation(t *testing.T) {
 	}
 	for i, test := range tests {
 		err := ValidateTreeForCreation(test.tree)
-		if hasErr := err != nil; hasErr != test.wantErr {
+		switch hasErr := err != nil; {
+		case hasErr != test.wantErr:
 			t.Errorf("%v: ValidateTreeForCreation() = %v, wantErr = %v", i, err, test.wantErr)
+		case hasErr && errors.ErrorCode(err) != errors.InvalidArgument:
+			t.Errorf("%v: ValidateTreeForCreation() = %v, wantCode = %v", i, err, errors.InvalidArgument)
 		}
```

Comment:
I found it strange at first, but I find that I prefer switch over "if/else if" chains now, as the initialization gets a separate line from the conditions.